spec/lib/whitehall_importer/embed_body_references_spec.rb

```diff
@@ -12,49 +12,106 @@
-      govspeak_body = described_class.call(
+    it "converts Whitehall image bang embeds to govspeak Image embeds" do
+      body = described_class.call(body: "!!1", images: ["file.jpg"])
+
+      expect(body).to eq("[Image:file.jpg]")
+    end
+
+    it "converts Whitehall attachment bang embeds to govspeak Attachment embeds" do
+      body = described_class.call(body: "!@1", attachments: ["file.pdf"])
+
+      expect(body).to eq("[Attachment:file.pdf]")
+    end
+
+    it "ignore Whitehall image bang embeds that are neither start of the string or preceeded by new lines" do
+      body = described_class.call(
         body: "!!1 test !!2",
-        images: ["foo.png", "bar.jpg"],
+        images: ["file.png", "file.jpg"],
       )
 
-      expect(govspeak_body).to eq("[Image:foo.png] test [Image:bar.jpg]")
+      expect(body).to eq("[Image:file.png] test !!2")
     end
 
-    it "removes any image markdown that doesn't resolve to an image" do
-      govspeak_body = described_class.call(
-        body: "Bar !!2 Baz",
-        images: ["foo.png"],
+    it "ignore Whitehall attachment bang embeds that are neither start of the string or preceeded by new lines" do
+      body = described_class.call(
+        body: "!@1 test !@2",
+        attachments: ["file.pdf", "file.csv"],
       )
 
-      expect(govspeak_body).to eq("Bar  Baz")
+      expect(body).to eq("[Attachment:file.pdf] test !@2")
     end
 
-    it "converts Whitehall attachment syntax to Content Publisher syntax" do
-      govspeak_body = described_class.call(
-        body: "!@1 test !@2",
-        attachments: ["file.pdf", "download.csv"],
+    it "ignore any image markdown that doesn't resolve to an image" do
+      body = described_class.call(
+        body: "Bar !!2 Baz",
+        images: ["file.jpg"],
       )
 
-      expect(govspeak_body).to eq("[Attachment:file.pdf] test [Attachment:download.csv]")
+      expect(body).to eq("Bar !!2 Baz")
     end
 
-    it "converts Whitehall inline attachment syntax to Content Publisher syntax" do
-      govspeak_body = described_class.call(
-        body: "[InlineAttachment:1] test [InlineAttachment:2]",
-        attachments: ["file.pdf", "download.csv"],
+    it "ignore any attachment markdown that doesn't resolve to an attachment" do
+      body = described_class.call(
+        body: "Bar !@2 Baz",
+        attachments: ["file.pdf"],
       )
 
-      expect(govspeak_body).to eq("[AttachmentLink:file.pdf] test [AttachmentLink:download.csv]")
+      expect(body).to eq("Bar !@2 Baz")
+    end
+
+    it "converts image and attachment embeds prefixed by single new lines to be prefixed by two" do
+      prefix_conversions = [
+        { prefix: "\n", conversion: "\n\n" },
+        { prefix: "\r\n", conversion: "\r\n\r\n" },
+      ]
+
+      prefix_conversions.each do |prefix_conversion|
+        prefixed = []
+        prefixed << "#{prefix_conversion[:prefix]}!!1"
+        prefixed << "#{prefix_conversion[:prefix]}!@1"
```

Comment:
Minor but could do this more directly and concisely with: 

```
        prefixed = ["#{prefix_conversion[:prefix]}!!1",
"#{prefix_conversion[:prefix]}!@1"]
```

Reply:
Agree.